packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -91,34 +100,170 @@ class BottomNavigationBar extends StatefulWidget {
+      child: new Container(
+        margin: new EdgeInsets.only(
+          top: new Tween<double>(
+            begin: type == BottomNavigationBarType.fixed ? 8.0 : 16.0,
+            end: _kTopMargin,
+          ).evaluate(animation),
+        ),
+        child: new IconTheme(
+          data: new IconThemeData(
+            color: type == BottomNavigationBarType.fixed ? colorTween.evaluate(animation) : Colors.white,
+            size: iconSize,
+          ),
+          child: item.icon,
+        ),
+      ),
+    );
+  }
+
+  Widget _buildFixedLabel() {
+    return new Align(
+      alignment: Alignment.bottomCenter,
+      heightFactor: 1.0,
+      child: new Container(
+        margin: const EdgeInsets.only(bottom: _kBottomMargin),
+        child: DefaultTextStyle.merge(
+          style: new TextStyle(
+            fontSize: _kActiveFontSize,
+            color: colorTween.evaluate(animation),
+          ),
+          // The font size should grow here when active, but because of the way
+          // font rendering works, it doesn't grow smoothly if we just animate
+          // the font size, so we use a transform instead.
+          child: new Transform(
+            transform: new Matrix4.diagonal3(
+              new Vector3.all(
+                new Tween<double>(
+                  begin: _kInactiveFontSize / _kActiveFontSize,
+                  end: 1.0,
+                ).evaluate(animation),
+              ),
+            ),
+            alignment: Alignment.bottomCenter,
+            child: item.title,
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildShiftingLabel() {
+    return  new Align(
+      alignment: Alignment.bottomCenter,
+      heightFactor: 1.0,
+      child: new Container(
+        margin: new EdgeInsets.only(
+          bottom: new Tween<double>(
+            // In the spec, they just remove the label for inactive items and
+            // specify a 16dp bottom margin. We don't want to actually remove
+            // the label because we want to fade it in and out, so this modifies
+            // the bottom margin to take that into account.
+            begin: 2.0,
+            end: _kBottomMargin,
+          ).evaluate(animation),
+        ),
+        child: new FadeTransition(
+          opacity: animation,
+          child: DefaultTextStyle.merge(
+              style: const TextStyle(
+                fontSize: _kActiveFontSize,
+                color: Colors.white,
+              ),
+              child: item.title,
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // In order to use the flex container to grow the tile during animation, we
+    // need to divide the changes in flex allotment into smaller pieces to
+    // produce smooth animation. We do this by multiplying the flex value
+    // (which is an integer) by a large number.
+    final int size = type == BottomNavigationBarType.shifting ? (flex * 1000.0).round() : 1;
```

Comment:
prefer using a switch

Using a switch for enums means that if we ever add a value to the enum, the analyzer will immediately flag every place we need to update. It's a bit more verbose, but the analyzer benefits are worth it. We do this consistently for all enums because that's an easier rule to follow that having exceptions for some and not others.